src/store/modules/Auth.ts

```diff
@@ -0,0 +1,40 @@
+import { Module } from 'vuex'
+import { AuthState } from '@/types/Vuex'
+import { User } from '@/types/User'
+
+const Auth: Module<AuthState, any> = {
+  state: {
+    // TODO: initial user state should be null
+    // send a request to check if the user is logged in
+    // if yes, set the user state
+    user: null,
+    token: null
+  },
+  getters: {
+    user(state) {
+      return state.user
+    },
+    userId(state) {
+      return state.user?._id
+    },
+    token(state) {
+      return state.token
+    },
+    isLoggedIn(state) {
+      return !!state.user
+    }
+  },
+  mutations: {
+    setUser(state, user: User) {
+      state.user = user
+    }
+  },
+  actions: {
+    async login({ commit }, { email, password }) {
+      // HTTP request to login
+      // commit('setUser', user)
+    }
+  }
+}
+
+export default Auth
```

Comment:
This may not occur, but given the AWS module for the sign in is also called Auth there is potential for some collision there

Reply:
There is no need to worry about conflicts because the namespace of what is being exported is defined only by the import statement. ie if i export default Auth, I can go to the file that needs to use the Vuex Auth and type import VuexAuth from 'file/path' then type import AWSAuth from 'file/path'. Now both can be used and are easily distinguishable